web/frontend/src/pages/election/Result.tsx

```diff
@@ -0,0 +1,168 @@
+  // are mapped to 0 or 1s, such that ballots can be count more efficiently
+  const groupByID = (
+    resultMap: Map<any, any>,
+    IDs: ID[],
+    results: any,
+    toNumber: boolean = false
+  ) => {
+    IDs.forEach((id, index) => {
+      let updatedRes: number[][] = new Array<number[]>();
+      let res: number[] = results[index];
+
+      if (toNumber) {
+        res = results[index].map((r: boolean) => (r ? 1 : 0));
+      }
+
+      if (resultMap.has(id)) {
+        updatedRes = resultMap.get(id);
+      }
+
+      updatedRes.push(res);
+      resultMap.set(id, updatedRes);
+    });
+  };
+
+  const groupResultsByID = () => {
+    const selectRes: SelectResults = new Map<ID, number[][]>();
+    const rankRes: RankResults = new Map<ID, number[][]>();
+    const textRes: TextResults = new Map<ID, string[][]>();
+
+    result.forEach((res) => {
+      groupByID(selectRes, res.SelectResultIDs, res.SelectResult, true);
+      groupByID(rankRes, res.RankResultIDs, res.RankResult);
+      groupByID(textRes, res.TextResultIDs, res.TextResult);
+    });
+
+    return { rankRes, selectRes, textRes };
+  };
+
+  useEffect(() => {
+    if (result !== null) {
+      const { rankRes, selectRes, textRes } = groupResultsByID();
+
+      setRankResult(rankRes);
+      setSelectResult(selectRes);
+      setTextResult(textRes);
+    }
+  }, [result]);
+
+  const SubjectElementResultDisplay = (element: SubjectElement) => {
+    return (
+      <div className="px-4 pb-4">
+        <h2 className="text-lg pb-2">{element.Title}</h2>
+        {element.Type === RANK && (
+          <RankResult rank={element as RankQuestion} rankResult={rankResult.get(element.ID)} />
+        )}
+        {element.Type === SELECT && (
+          <SelectResult
+            select={element as SelectQuestion}
+            selectResult={selectResult.get(element.ID)}
+          />
+        )}
+        {element.Type === TEXT && <TextResult textResult={textResult.get(element.ID)} />}
+      </div>
+    );
+  };
+
+  const displayResults = (subject: Subject) => {
+    return (
+      <div key={subject.ID} className="px-8 pt-4">
+        <h2 className="text-lg font-bold">{subject.Title}</h2>
+        {subject.Order.map((id: ID) => (
+          <div key={id}>
+            {subject.Elements.get(id).Type === SUBJECT ? (
+              <div className="px-4">{displayResults(subject.Elements.get(id) as Subject)}</div>
+            ) : (
+              SubjectElementResultDisplay(subject.Elements.get(id))
+            )}
+          </div>
+        ))}
+      </div>
+    );
+  };
+
+  const exportData = () => {
+    const fileName = 'result.json';
+
+    // Create a blob of the data
+    const fileToSave = new Blob([JSON.stringify({ Result: result })], {
+      type: 'application/json',
+    });
+
+    saveAs(fileToSave, fileName);
+    //https://stackoverflow.com/questions/19721439/download-json-object-as-a-file-from-browser
```

Comment:
```suggestion
    // https://stackoverflow.com/a/45594892
    saveAs(fileToSave, fileName);
```

(SO answers have a convenient "share" button that provides shorter URL ;) )

Reply:
Haha true, admittedly I only copy pasted the code and the link that was previously in DownloadResult.tsx 😁